File: sliding-window/max-vowels.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        count = sum(1 for c in s[:k] if c in vowels)
        maxV = count
        for i in range(k,len(s)):
            if s[i-k] in vowels:
                count -=1
            if s[i] in vowels:
                count += 1
            maxV = max(maxV, count)
        return maxV

File: sliding-window/max-vowels.go
func maxVowels(s string, k int) int {
	isVowel := [256]bool{}
	for _, v := range "aeiou" {
		isVowel[v] = true
	}

	var total, maxV int
	for i := 0; i < k; i++ {
		if isVowel[s[i]] {
			total++
		}
	}

	maxV = total
	for i := k; i < len(s); i++ {
		if isVowel[s[i-k]] {
			total--
		}
		if isVowel[s[i]] {
			total++
		}
		if total > maxV {
			maxV = total
		}
	}
	return maxV
}